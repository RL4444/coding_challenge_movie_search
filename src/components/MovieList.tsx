import React, { FC, ReactElement, useState } from 'react';
import styled from 'styled-components';

import useApi from '../hooks/useApi';

import Spinner from './Spinner';
import Movie from './Movie';

import { IApiResponse } from '../types/movie';

interface IListProps {
    searchTerm: string;
}

const SearchTermText = styled.p`
    margin-top: 2em;
    margin-bottom: 1em;
    font-size: 22px;
    font-family: ${(p) => p.theme.fonts.title};
    color: ${(p) => p.theme.colors.white};
    text-align: left;
`;

const List: FC<IListProps> = ({ searchTerm }) => {
    const { response, error, loading }: IApiResponse = useApi(searchTerm);

    console.log('response ', response);

    return (
        <>
            <SearchTermText>
                {!loading && response && response.Error
                    ? `No results for for "${searchTerm}". Try defining your search more by adding more words.`
                    : response?.Search && `Results for "${searchTerm}"`}
            </SearchTermText>
            {!error &&
                !loading &&
                response?.Search &&
                response.Search.map(({ Poster, Title, Type, Year, imdbID }) => {
                    return <Movie key={imdbID} Poster={Poster} Title={Title} Type={Type} Year={Year} imdbID={imdbID} />;
                })}
        </>
    );
};

export default List;
