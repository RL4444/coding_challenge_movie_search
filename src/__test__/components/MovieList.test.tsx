import React from 'react';
import { shallow } from 'enzyme';

import MovieList from '../../components/MovieList';

it('renders without crashing with an empty string', () => {
    const component = shallow(<MovieList searchTerm={''} />);
    expect(component).toMatchSnapshot();
});
