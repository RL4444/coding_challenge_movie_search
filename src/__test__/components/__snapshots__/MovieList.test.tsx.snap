// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing with an empty string 1`] = `
<Fragment>
  <styled.p />
</Fragment>
`;
